### PR TITLE
Add docstrings for views

### PR DESCRIPTION
**Pull request by KevinEtienne**
Adding docstrings for view would allow to use the browsable api from the `djangorestframework`

### CODE REVIEW

File: user_management/api/views.py [hunk 1]
```diff
@@ -15,6 +15,15 @@
 
 
 class GetAuthToken(ObtainAuthToken):
+    """
+    Authentication by token.
+
+    Define `POST` (create) method to authenticate a user from its `email` and
```
Comment (Ian-Foote) [inline user_management/api/views.py]: ~~`from its`~~ `using their`

File: user_management/api/views.py [hunk 2]
```diff
@@ -15,6 +15,15 @@
...
+    `password` and return a `token` if successful.
+    The `token` would be valid for a value defined in `settings.AUTH_TOKEN_MAX_AGE`.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: The token remains valid until `settings.AUTH_TOKEN_MAX_AGE` time has passed.

File: user_management/api/views.py [hunk 3]
```diff
@@ -15,6 +15,15 @@
...
+
+    `DELETE` method remove the current `token` from the database.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: `removes`

File: user_management/api/views.py [hunk 4]
```diff
@@ -59,6 +68,11 @@ def delete(self, request, *args, **kwargs):
 
 
 class UserRegister(generics.CreateAPIView):
+    """
+    User registration.
+
+    Register a user and send an email to validate the new account.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: I think the `User` can be set up to not require email validation. In this case `UserRegister

File: user_management/api/views.py [hunk 5]
```diff
@@ -59,6 +68,11 @@ def delete(self, request, *args, **kwargs):
 
 
 class UserRegister(generics.CreateAPIView):
+    """
+    User registration.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: `Register a new User.`

File: user_management/api/views.py [hunk 6]
```diff
@@ -95,6 +109,12 @@ def is_valid(self, serializer):
 
 
 class PasswordResetEmail(generics.GenericAPIView):
+    """
+    Password reset request.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Send a password reset email to a user on request.

File: user_management/api/views.py [hunk 7]
```diff
@@ -95,6 +109,12 @@ def is_valid(self, serializer):
...
+
+    Accept an `email` and send an email to reset a password if the user is found.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: A user can request a password request email by providing their email address.

File: user_management/api/views.py [hunk 8]
```diff
@@ -122,6 +142,11 @@ def post(self, request, *args, **kwargs):
 
 
 class OneTimeUseAPIMixin(object):
+    """
+    Check `uid` and `token`.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Use a `uid` and a `token` to allow one-time access to a view.

File: user_management/api/views.py [hunk 9]
```diff
@@ -143,6 +168,17 @@ def initial(self, request, *args, **kwargs):
 
 
 class PasswordReset(OneTimeUseAPIMixin, generics.UpdateAPIView):
+    """
+    Password reset update view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Allow a user to reset their password.

File: user_management/api/views.py [hunk 10]
```diff
@@ -152,6 +188,12 @@ def get_object(self):
 
 
 class PasswordChange(generics.UpdateAPIView):
+    """
+    Password update view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Change a user's password.

File: user_management/api/views.py [hunk 11]
```diff
@@ -161,6 +203,11 @@ def get_object(self):
 
 
 class VerifyAccountView(OneTimeUseAPIMixin, views.APIView):
+    """
+    Verify account view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Verify a new user's email address.

File: user_management/api/views.py [hunk 12]
```diff
@@ -179,6 +226,11 @@ def post(self, request, *args, **kwargs):
 
 
 class ProfileDetail(generics.RetrieveUpdateDestroyAPIView):
+    """
+    User profile detail view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Allow a user to view and edit their profile information.

File: user_management/api/views.py [hunk 13]
```diff
@@ -15,6 +15,15 @@
 
 
 class GetAuthToken(ObtainAuthToken):
+    """
+    Authentication by token.
```
Comment (meshy) [inline user_management/api/views.py]: Obtain an authentication token.

File: user_management/api/views.py [hunk 14]
```diff
@@ -188,19 +240,33 @@ def get_object(self):
 
 
 class UserList(generics.ListCreateAPIView):
+    """
+    User list and create view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Return information about all users and allow creation of new users.

File: user_management/api/views.py [hunk 15]
```diff
@@ -188,19 +240,33 @@ def get_object(self):
...
+    Allow to `GET` a list users and to `POST` new user for admin user only.
+    """
     model = User
     permission_classes = (IsAuthenticated, permissions.IsAdminOrReadOnly)
     serializer_class = serializers.UserSerializerCreate
 
 
 class UserDetail(generics.RetrieveUpdateDestroyAPIView):
+    """
+    User detail view.
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Display information about a user.

Allow admin users to update or delete user information.

File: user_management/api/views.py [hunk 16]
```diff
@@ -15,6 +15,16 @@
...
 
 class GetAuthToken(ObtainAuthToken):
+    """
+    Obtain an authentication token.
+
+    Define a `POST` (create) method to authenticate a user from useing their `email` and
+    `password` and return a `token` if successful.
+    The `token` remains valid until `settings.AUTH_TOKEN_MAX_AGE` time has passed.
+
+
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Stray blank line here.

File: user_management/api/views.py [hunk 17]
```diff
@@ -122,6 +144,12 @@ def post(self, request, *args, **kwargs):
 
 
 class OneTimeUseAPIMixin(object):
+    """
+    Use a `uid` and a `token` to allow one-time access to a view.
+
+
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Another stray line.

File: user_management/api/views.py [hunk 18]
```diff
@@ -188,19 +241,33 @@ def get_object(self):
...
+    Allow admin users to update or delete user information.
+    """
     model = User
     permission_classes = (IsAuthenticated, permissions.IsAdminOrReadOnly)
     serializer_class = serializers.UserSerializer
 
 
 class ResendConfirmationEmail(generics.GenericAPIView):
-    """Resend a confirmation email."""
+    """Resend a confirmation email.
```
Comment (meshy) [inline user_management/api/views.py]: Quotes on new line for multi-line docstrings, please

**Comment by meshy** [comment]
Looks good `:)`

**Comment by KevinEtienne** [comment]
Docstrings for `avatar` were all ready done: https://github.com/incuna/django-user-management/blob/ed8e647586c5a9ca48e9223aa19ab99d545c3167/user_management/api/avatar/views.py

**Comment by meshy** [comment]
:+1:

**Pull request merged**